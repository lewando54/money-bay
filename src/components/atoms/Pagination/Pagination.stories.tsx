/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import Pagination from './Pagination'

const meta = {
	component: Pagination,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof Pagination>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
