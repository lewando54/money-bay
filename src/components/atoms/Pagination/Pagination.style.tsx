import { Dimensions, StyleSheet } from 'react-native'

const {width} = Dimensions.get('window')

export default StyleSheet.create({
    container: {
        flexDirection: 'row',
        marginLeft: 'auto',
        marginRight: 'auto',
        justifyContent: 'space-between',
        width: width * 0.128
    }
})
