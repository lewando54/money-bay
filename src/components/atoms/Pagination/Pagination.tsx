import React from 'react'
import paginationStyle from './Pagination.style'
import { TOnboardingSlide } from '../../organisms/OnboardingSlides/OnboardingSlides'
import { SharedValue } from 'react-native-reanimated'
import { View } from 'react-native'
import Dot from '../Dot/Dot'
import { MAIN_DARK } from '../../../styling/GlobalTheme.style'

interface IPaginationProps {
    data?: TOnboardingSlide
    x?: SharedValue<number>
    color?: string
    testID?: string
}

export default function Pagination({
    data=[{id: 0, imageSource: require('../../../../assets/cardIssuers/mastercard.png'), heading: 'test', body: 'test', testID: 'testid'}],
    x,
    color=MAIN_DARK,
    testID
}: IPaginationProps){
    return (
        <View testID={testID} style={paginationStyle.container}>
            {data.map((_, index) => {
                return <Dot key={index} index={index} x={x} color={color}/>
            })}
        </View>
    )
}
