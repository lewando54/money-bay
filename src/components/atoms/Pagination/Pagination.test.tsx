import React from 'react'
import renderer from 'react-test-renderer'

import Pagination from './Pagination'
describe('Pagination', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Pagination />)
        expect(tree).toMatchSnapshot()
    })
})
