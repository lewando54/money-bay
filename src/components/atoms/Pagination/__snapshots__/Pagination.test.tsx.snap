// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginLeft": "auto",
      "marginRight": "auto",
      "width": 96,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "borderColor": "#1B1D4D",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 8,
        "width": 8,
      }
    }
  />
</View>
`;
